sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/implementation/ModelsRepositoryAPIImpl.java

```diff
@@ -0,0 +1,114 @@
+public final class ModelsRepositoryAPIImpl {
+    /** server parameter. */
+    private final String host;
+
+    /**
+     * Gets server parameter.
+     *
+     * @return the host value.
+     */
+    public String getHost() {
+        return this.host;
+    }
+
+    /** Api Version. */
+    private final String apiVersion;
+
+    /**
+     * Gets Api Version.
+     *
+     * @return the apiVersion value.
+     */
+    public String getApiVersion() {
+        return this.apiVersion;
+    }
+
+    /** The HTTP pipeline to send requests through. */
+    private final HttpPipeline httpPipeline;
+
+    /**
+     * Gets The HTTP pipeline to send requests through.
+     *
+     * @return the httpPipeline value.
+     */
+    public HttpPipeline getHttpPipeline() {
+        return this.httpPipeline;
+    }
+
+    /** The serializer to serialize an object into a string. */
+    private final SerializerAdapter serializerAdapter;
+
+    /**
+     * Gets The serializer to serialize an object into a string.
+     *
+     * @return the serializerAdapter value.
+     */
+    public SerializerAdapter getSerializerAdapter() {
+        return this.serializerAdapter;
+    }
+
+    /** The ModelsRepositoryImpl object to access its operations. */
+    private final ModelsRepositoryImpl modelsRepository;
+
+    /**
+     * Gets the ModelsRepositoryImpl object to access its operations.
+     *
+     * @return the ModelsRepositoryImpl object.
+     */
+    public ModelsRepositoryImpl getModelsRepository() {
+        return this.modelsRepository;
+    }
+
+    /**
+     * Initializes an instance of ModelsRepositoryAPI client.
+     *
+     * @param host server parameter.
+     */
+    ModelsRepositoryAPIImpl(String host) {
+        this(
+                new HttpPipelineBuilder()
+                        .policies(new UserAgentPolicy(), new RetryPolicy(), new CookiePolicy())
+                        .build(),
+                JacksonAdapter.createDefaultSerializerAdapter(),
+                host);
+    }
+
+    /**
+     * Initializes an instance of ModelsRepositoryAPI client.
+     *
+     * @param httpPipeline The HTTP pipeline to send requests through.
+     * @param host server parameter.
+     */
+    ModelsRepositoryAPIImpl(HttpPipeline httpPipeline, String host) {
+        this(httpPipeline, JacksonAdapter.createDefaultSerializerAdapter(), host);
+    }
+
+    /**
+     * Initializes an instance of ModelsRepositoryAPI client.
+     *
+     * @param httpPipeline The HTTP pipeline to send requests through.
+     * @param serializerAdapter The serializer to serialize an object into a string.
+     * @param host server parameter.
+     */
+    ModelsRepositoryAPIImpl(HttpPipeline httpPipeline, SerializerAdapter serializerAdapter, String host) {
+        this.httpPipeline = httpPipeline;
+        this.serializerAdapter = serializerAdapter;
+        this.host = host;
+        this.apiVersion = "2020-10-31";
```

Comment:
Why is it hard coded to this API version?

Reply:
excellent catch ! fixing.